src/clients/contact-client.js

```diff
@@ -0,0 +1,54 @@
+'use strict'
+
+const Client = require('./base-client')
+
+class ContactClient {
+  constructor (url) {
+    this.client = new Client({
+      name: 'ContactClient',
+      url: `${url}/api/contacts`,
+      contentType: 'application/vnd.api+json'
+    })
+  }
+
+  search (parameters = {}) {
+    return this.client.search(parameters)
+  }
+
+  find (id, parameters = {}) {
+    return this.client.find(id, parameters)
+  }
+
+  create (data, parameters = {}) {
+    return this.client.create(data, parameters)
+  }
+
+  update (id, data, parameters = {}) {
+    return this.client.update(id, data, parameters)
+  }
+
+  destroy (id, parameters = {}) {
+    return this.client.destroy(id, parameters)
+  }
+
+  uploadAvatar (id, data) {
+    return this.client.sendRequest(`${this.client.baseUrl}/${id}/avatar`, {
+      body: data,
+      method: 'POST'
+    })
+  }
+
+  deleteAvatar (id) {
+    return this.client.sendRequest(`${this.client.baseUrl}/${id}/avatar`, {
+      method: 'DELETE'
+    })
+  }
+
+  toggleGravatar (id) {
+    return this.client.sendRequest(`${this.client.baseUrl}/${id}/toggle_gravatar`, {
+      method: 'PUT'
```

Comment:
Just please double check whenever we have PUT/PATCH here. Probably it's PUT.

Reply:
https://github.com/infopark-customers/bima-meinebima2/blob/42ef9a711b6ca4f983332f389ef6a3d936038fbc/client/src/stores/contact/store.js#L149 ;)